### ISSUE TITLE
Inconsistent results between 16 and 64 MPI ranks

### ISSUE THREAD

**Issue by Thomas-Ulrich** [SeisSol/SeisSol#1181]
**Describe the bug**
on the master branch (5e75cf7) as well as older branches (v1.1.3), a specific setup with 16mio cells diverges when using 64 ranks, and run smoothly (final Mw~1.3) on 16 ranks.
on v1.1.2, no bug.

**Expected behavior**
same behaviour independent of number of ranks.

**To Reproduce**

seissol installed with:
```
spack install -j 40 seissol@1.1.3 convergence_order=4 dr_quad_rule=dunavant equations=elastic precision=single %intel  ^easi +python
```
on supermuc NG

setup available at /hppfs/work/pn49ha/di73yeq4/setup_francesco

(see also bashScript_NG_16ranks.sh  bashScript_NG_64ranks.sh for launch scripts).

EDIT: tested commit merging https://github.com/SeisSol/SeisSol/pull/982: error is already there

**Comment by Thomas-Ulrich** [comment]
This is the fault slip on 16 ranks
![image](https://github.com/user-attachments/assets/ab0ee7b6-85ff-4329-8ba4-b89808258357)
and on 64 ranks 
![image](https://github.com/user-attachments/assets/0e4773dd-84b4-4558-bfad-4731e92c1ef9)
with v1.0.1 on superNG

**Comment by Thomas-Ulrich** [comment]
The boundary conditions looks good
![image](https://github.com/user-attachments/assets/69ffc0ab-cbb0-493d-8a27-536d744273a1)

**Comment by davschneller** [comment]
That is very strange. By chance—do you have a clustering file both of a 16-rank and a 64-rank run at hand?

**Comment by davschneller** [comment]
Two things would come to mind as well:

* does it happen with GTS?
* if it's not too much work: does it happen before the actor model was introduced? (i.e. with tag sc21 or the likes)

**Comment by davschneller** [comment]
Not 100%ly sure if it helps, but would maybe https://github.com/SeisSol/SeisSol/tree/davschneller/cluster-mpi-problem-fix-attempt help?

I'm maybe getting a bit paranoid right now; but the case of having ghost regions with different clusters may not be handled properly; e.g. they may not present in the data regions (and especially the tree storage) at all.

Maybe there's also a relation to #839 .

**Comment by Thomas-Ulrich** [comment]
it seems to fix the problem!

**Comment by davschneller** [comment]
That... might be just an (at least) 8-year-old bug then.

**Comment by davschneller** [comment]
If it works now, then #1204 implements the fix (the branch up there had one unnecssary change in).

Though it may be good to try it multiple times (if you haven't done that already)—especially if it is a clustering/partitioning issue.

**Comment by Thomas-Ulrich** [comment]
Yes, I will test it again and then we can merge.
Sorry about the delay, and thanks again for fixing this bug!
(By the way, maybe now the CI/CD with multiple ranks will work?)

**Comment by Thomas-Ulrich** [comment]
I run it multiple times, and I never get the bug using davschneller/more-local-clusters

**Comment by Thomas-Ulrich** [comment]
closed by #1204
